src/containers/CodesplainAppBar.jsx

```diff
@@ -73,6 +75,19 @@ const styles = {
   },
 };
 
+const orderSnippets = (username, userSnippets, orgSnippets) => {
+  const allSnippets = { ...orgSnippets }; // Spread so there aren't side effects.
+  allSnippets[username] = userSnippets;
+  const orderedSnippets = _.reduce(allSnippets, (allList, snippets, role) => {
+    const next = _.reduce(snippets, (snipList, { language, lastEdited }, snippetTitle) => {
+      const obj = { role, snippetTitle, language, lastEdited };
+      return _.concat(snipList, obj);
+    }, []);
+    return _.concat(allList, next);
+  }, []);
+  return _.sortBy(orderedSnippets, ({ snippetTitle }) => snippetTitle);
+};
```

Comment:
I'll add a comment, it builds a vector out of the nested objects that contains only the details relevant to the search menu, and is sortable by name.